jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/criteria/TestCriteriaBuilder.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2018 IBM Corporation. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     10/01/2018: Will Dazey
+//       - #253: Add support for embedded constructor results with CriteriaBuilder
+package org.eclipse.persistence.jpa.test.criteria;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CompoundSelection;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Join;
+import javax.persistence.criteria.Root;
+
+import org.eclipse.persistence.jpa.test.criteria.model.L1;
+import org.eclipse.persistence.jpa.test.criteria.model.L1Model;
+import org.eclipse.persistence.jpa.test.criteria.model.L1_;
+import org.eclipse.persistence.jpa.test.criteria.model.L2;
+import org.eclipse.persistence.jpa.test.criteria.model.L2Model;
+import org.eclipse.persistence.jpa.test.criteria.model.L2_;
+import org.eclipse.persistence.jpa.test.framework.DDLGen;
+import org.eclipse.persistence.jpa.test.framework.Emf;
+import org.eclipse.persistence.jpa.test.framework.EmfRunner;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(EmfRunner.class)
+public class TestCriteriaBuilder {
+
+    @Emf(createTables = DDLGen.DROP_CREATE, classes = { L1.class, L2.class })
+    private EntityManagerFactory emf;
+
+    /**
+     * Merging ElementCollections on Oracle fails when EclipseLink generates 
+     * a DELETE SQL statement with a WHERE clause containing a CLOB.
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testCriteriaCompoundSelectionModel() throws Exception {
+        //Populate the database
+        EntityManager em = emf.createEntityManager();
+
+        em.getTransaction().begin();
+
+        L2 l2 = new L2(1, "L2-1");
+        L2 l2_2 = new L2(2, "L2-1");
```

Comment:
don't you want to have 'L2-2' here? Or is the same string in 'l2' and 'l2_2' intentional?

Reply:
Ya... I noticed this copy-paste mistake and figured since it doesn't change the test, it wasn't important enough to fix. If I'm already in there making changes tho, Ill fix this :(